app/order/workflows.go

```diff
@@ -178,15 +184,31 @@ func (wf *orderImpl) cancelUnavailableFulfillments() {
 }
 
 func (wf *orderImpl) waitForCustomer(ctx workflow.Context) (string, error) {
+	var signal CustomerActionSignal
+
+	s := workflow.NewSelector(ctx)
```

Comment:
While this works and is probably the best way, if you did want to avoid a selector, you could `workflow.AwaitWithTimeout(ctx, customerActionTimeout, func() bool { return ch.Len() > 0 })` and use whether response is true or false to choose code path.

Reply:
That's a pattern I'd not seen before. I think given we have no other Selector use in the app we should keep it as it is. Would be nice to see Await somewhere as well, but I think Selector is more important to show, coupled with arguably being the more idiomatic way.